ga4gh/datamodel/sequenceAnnotations.py

```diff
@@ -76,35 +76,65 @@ def __init__(self, dbFile):
         except without the pagetoken/size.
         """
-        # TODO: Refactor out common bits of this and the below search query.
-        sql = ("SELECT COUNT(*) FROM FEATURE WHERE "
-               "reference_name = ? "
-               "AND end > ? "  # compare this to query start
-               "AND start < ? "  # and this to query end
-               )
+        _, sql_args, sql = self.featuresQuery(
+            pageToken=None, pageSize=None,
+            referenceName=referenceName, start=start, end=end,
+            parentId=parentId, featureTypes=featureTypes,
+            name=name, geneSymbol=geneSymbol)
+        query = self._dbconn.execute(sql, sql_args)
+        return (query.fetchone())[0]
+
+    def featuresQuery(self, **kwargs):
+        """
+        Converts a dictionary of keyword arguments into a tuple
+        of SQL select statements, the list of SQL arguments, and
+        a SQL count statement.
+        """
         # TODO: Optimize by refactoring out string concatenation
-        sql_args = (referenceName, start, end)
-        if parentId is not None:
+        sql = ""
+        sql_rows = "SELECT * FROM FEATURE WHERE id > 1 "
+        sql_count = "SELECT COUNT(*) FROM FEATURE WHERE id > 1 "
+        sql_args = ()
+        if 'name' in kwargs and kwargs['name']:
+            sql += "AND name = ? "  # compare this to query start
+            sql_args += (kwargs.get('name'),)
+        if 'geneSymbol' in kwargs and kwargs['geneSymbol']:
+            sql += "AND gene_name = ? "  # compare this to query start
+            sql_args += (kwargs.get('geneSymbol'),)
+        if 'start' in kwargs and kwargs['start'] is not None:
+            sql += "AND end > ? "  # compare this to query start
+            sql_args += (kwargs.get('start'),)
+        if 'end' in kwargs and kwargs['end'] is not None:
+            sql += "AND start < ? "  # and this to query end
+            sql_args += (kwargs.get('end'),)
+        if 'referenceName' in kwargs and kwargs['referenceName']:
+            sql += "AND reference_name = ?"
+            sql_args += (kwargs.get('referenceName'),)
+        if 'parentId' in kwargs and kwargs['parentId']:
             sql += "AND parent_id = ? "
-            sql_args += (parentId,)
-        if featureTypes is not None and len(featureTypes) > 0:
+            sql_args += (kwargs['parentId'],)
+        if kwargs.get('featureTypes') is not None \
+                and len(kwargs['featureTypes']) > 0:
             sql += "AND type IN ("
-            sql += ", ".join(["?", ] * len(featureTypes))
+            sql += ", ".join(["?", ] * len(kwargs.get('featureTypes')))
             sql += ") "
-            sql_args += tuple(featureTypes)
-        query = self._dbconn.execute(sql, sql_args)
-        return (query.fetchone())[0]
+            sql_args += tuple(kwargs.get('featureTypes'))
+        sql_rows += sql
+        sql_count += sql
+        return sql_rows, sql_args, sql_count
```

Comment:
tiny quibble... wouldn't it be neater if the return list order were `sql_rows, sql_count, sql_args` (ie. string, string, list)